ddtrace/profiling/recorder.py

```diff
@@ -1,16 +1,35 @@
 # -*- encoding: utf-8 -*-
 import collections
 
-from ddtrace.profiling import _attr
 from ddtrace.vendor import attr
 
 
+class _defaultdictkey(dict):
+    """A variant of defaultdict that calls default_factory with the missing key as argument."""
+
+    def __init__(self, default_factory=None):
+        self.default_factory = default_factory
+
+    def __missing__(self, key):
+        if self.default_factory:
+            v = self[key] = self.default_factory(key)
+            return v
+        raise KeyError(key)
+
+
 @attr.s(slots=True, eq=False)
 class Recorder(object):
     """An object that records program activity."""
 
+    _DEFAULT_MAX_EVENTS = 32768
+
+    default_max_events = attr.ib(default=_DEFAULT_MAX_EVENTS)
+    """The default maximum number of events."""
```

Comment:
This docstring could maybe be clarified a tad. I initially interpreted this to be the overall maximum number of events by summing them all together. 

```suggestion
    """The maximum number of events for an event type if one is not specified."""
```